server/shaders/foveate.comp.glsl

```diff
@@ -0,0 +1,61 @@
+/*
+ * WiVRn VR streaming
+ * Copyright (C) 2024  galister <galister@librevr.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#version 450
+
+layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;
+
+layout(set = 0, binding = 0, rgba8) uniform writeonly restrict image2D target;
+
+layout(push_constant) uniform PushConstants
+{
+    vec2 a;
+    vec2 b;
+    vec2 scale;
+    vec2 center;
+} pcs;
+
+const float one_over_dim = 1.0 / 127.0;
```

Comment:
Can we have the dimension as a specialization constant and sync it with monado's image dimensions?